python/pyspark/sql/session.py

```diff
@@ -454,12 +484,12 @@ def getOrCreate(self) -> "SparkSession":
                                 RemoteSparkSession._start_connect_server(url, opts)
```

Comment:
Should fix above to:

```
                url = opts.get("spark.remote", os.environ.get("SPARK_REMOTE", None))
                if url is None:
                    raise RuntimeError("Spark Connect URL to the server was not found.")
```

too because now `SPARK_REMOTE` is optional.